### ISSUE TITLE
Error: ng g interface image => Invalid blueprint: interface

### ISSUE THREAD

**Issue by albanx** [angular/angular-cli#1225]
1. OS
**MAC OSX Yosemite 10.10.5**

2. NG Versions
```
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
angular-cli: 0.0.39
node: 4.1.1
os: darwin x64
```
3. Repro steps
`ng g interface image`

Just trying to generate an interface with the CLI. Not working.
Cheers
Thank you
Alban

**Comment by RicardoVaranda** [comment]
What exactly are you trying to create, interface is not a valid command the commands which you can use can be found here:

https://github.com/angular/angular-cli/tree/master/addon/ng2/blueprints

It seems like you want to create a visual component in which case you should first generate a route for it and then create and link in the directives a component for this.

**Comment by albanx** [comment]
I am trying to create a simple class Interface (not a visual stuff). Yes
interface is a valid command, and is in the above URL and in the guide
here:
https://github.com/angular/angular-cli#generating-components-directives-pipes-and-services


On Mon, Jun 27, 2016 at 4:23 PM, Ricardo Varanda <notifications@github.com>
wrote:

> What exactly are you trying to create, interface is not a valid command
> the commands which you can use can be found here:
>
> https://github.com/angular/angular-cli/tree/master/addon/ng2/blueprints
>
> It seems like you want to create a visual component in which case you
> should first generate a route for it and then create and link in the
> directives a component for this.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular-cli/issues/1225#issuecomment-228779095>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABFAqKd21Qg5IhRe9wBMQWbhwWROVCcQks5qP-rUgaJpZM4I_NMP>
> .
>



-- 
www.albanx.com
www.realuploader.com

**Comment by ciesielskico** [comment]
Your'e running an older version of angular-cli. Try updating.

https://github.com/angular/angular-cli#updating-angular-cli

**Comment by albanx** [comment]
Already tried with the above, and it installed the same version.

On Mon, Jun 27, 2016 at 4:30 PM, Alexander Ciesielski <
notifications@github.com> wrote:

> Your'e running an older version of angular-cli. Try updating.
>
> https://github.com/angular/angular-cli#updating-angular-cli
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular-cli/issues/1225#issuecomment-228781410>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABFAqHBMtUT1n-cFsQ0PwwN7r4Hg-OW3ks5qP-yOgaJpZM4I_NMP>
> .
>



-- 
www.albanx.com
www.realuploader.com

**Comment by albanx** [comment]
Ah wait, maybe I have to update the local one

On Mon, Jun 27, 2016 at 4:32 PM, Alban Xhaferllari <albanx@gmail.com> wrote:

> Already tried with the above, and it installed the same version.
>
> On Mon, Jun 27, 2016 at 4:30 PM, Alexander Ciesielski <
> notifications@github.com> wrote:
>
>> Your'e running an older version of angular-cli. Try updating.
>>
>> https://github.com/angular/angular-cli#updating-angular-cli
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/angular/angular-cli/issues/1225#issuecomment-228781410>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/ABFAqHBMtUT1n-cFsQ0PwwN7r4Hg-OW3ks5qP-yOgaJpZM4I_NMP>
>> .
>>
>
>
>
> --
> www.albanx.com
> www.realuploader.com
>



-- 
www.albanx.com
www.realuploader.com

**Comment by Splaktar** [comment]
@albanx This looks like a duplicate of #1224? Can you close one of them?

**Comment by albanx** [comment]
After updating to the last version I get this error:
```
alban-mac:public_web axhaferllari$ ng init
module.js:338
    throw err;
    ^

Error: Cannot find module 'exists-sync'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:286:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/export/www/alban/powerbast/public_web/node_modules/angular-cli/node_modules/ember-cli/lib/models/project.js:11:26)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
```

So seems redirecting to this one now https://github.com/angular/angular-cli/issues/1186

**Comment by albanx** [comment]
Fixed. Here it is what happened:

1. Update angular-cli following this guide https://github.com/angular/angular-cli#updating-angular-cli (both local and global). It updated to version angular-cli": "^1.0.0-beta.8". Tested and failed with the above error.
2. Then I went into node_modules/angular-cli and run npm install
3. Return to the root folder and run again npm install

This fixed my issue. Generating interface now is working
